File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for i in matrix:
            last = len(i)-1;
            if(target>i[last]):
                pass
            else:
                for j in i:
                    if(j==target):
                        return True
        return False

File: Data Structures & Algorithms/car-fleet/submission-0.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        int n = position.size();
        map<float,float,greater<float>> mp;
        stack<float> st;
        unsigned long result = 0;
        for(int i =0;i<n;i++){
            float dist = (target-position[i]);
            float spd = speed[i];
            float time = dist/spd;
            mp[position[i]]= time;
        }

        for(auto x: mp){
            cout<<x.first<<" "<<x.second<<endl;
            if(st.size() and st.top()<x.second){
                st.push(x.second);
            }
            else if (st.size()==0){
                st.push(x.second);
            }
            result = max(st.size(),result);
        }
        return result;
    }
};
